src/remoted/secure.c

```diff
@@ -272,6 +277,48 @@ void * rem_keyupdate_main(__attribute__((unused)) void * args) {
     }
 }
 
+// Closer rids thread
+static void * close_fp_main(void * args) {
+    keystore * keys = (keystore *)args;
+    int seconds;
+    int flag;
+
+    mdebug1("Rids closer thread started.");
+    seconds = logr.rids_closing_time;
+
+    while (1) {
+        sleep(seconds);
+        key_lock_write();
```

Comment:
This looks expensive and rather unnecessary. In my understaing:
- key_lock_read(): Some thread that is reading the keystore
- keyentries[i]->mutex: Mutual exclusion of that key entry. Can modifcate keyentry information in the process
- key_lock_write(): Block all reads to the keystore. Possible needed when keystore is reloaded or new keys are added

In my opinion only read lock is needed here. Also need to take the keyentry mutex before evaulating the file pointer